upload/system/library/myparcelnl/includes/class_myparcel_shipment_checkout.php

```diff
@@ -276,169 +275,90 @@ public function getDeliveryDateFromSavedData($myparcel_delivery_options)
-                    $fee = $this->convertPriceToFloat($fee);
-//                    $fee_including_tax = $this->getTotalDeliveryTaxAmountFromCart($fee, $cart);
-                    if ($addTax) {
-                        $fee_including_tax = $this->getTotalDeliveryTaxAmountFromCart($fee, $cart);
-                    } else {
-                        $fee_including_tax = $fee;
-                    }
-                    //$fee_including_tax = $fee;
-                    if ($price_format) {
-                        $formatted_fee = $this->formatDeliveryPrice($fee_including_tax, $currency, $color_format, $prefix);
-                    } else {
-                        $formatted_fee = $fee_including_tax;
-                    }
-                }
-
-                if ($price_type == 1 && (float)($checkout_settings['belgium_' . $option . '_fee2']) >= 0) {
-                    $fee = (float)$checkout_settings['belgium_' . $option . '_fee2'];
-                    $fee = $this->convertPriceToFloat($fee);
-//                    $fee_including_tax = $this->getTotalDeliveryTaxAmountFromCart($fee, $cart);
-                    if ($addTax) {
-                        $fee_including_tax = $this->getTotalDeliveryTaxAmountFromCart($fee, $cart);
-                    } else {
-                        $fee_including_tax = $fee;
-                    }
-                    //$fee_including_tax = $fee;
-                    if ($price_format) {
-                        $formatted_fee = $this->formatDeliveryPrice($fee_including_tax, $currency, $color_format, $prefix);
-                    } else {
-                        $formatted_fee = $fee_including_tax;
-                    }
-                }
-
-                $be_prices[$option] = $formatted_fee;
-            }
-
-            return array(
-                'NL' => $prices,
-                'BE' => $be_prices
-            );
-        }
-
-        return array(
-            'NL' => $prices,
-        );
-    }
+	/**
+	 * Retrieve delivery prices that are set in admin settings
+	 * @param boolean $priceFormat (add currency symbol and convert dot/comma)
+	 * @param boolean $colorFormat (add color code into price)
+	 * @param string $prefix prefix
+	 * @return array prices of delivery types
+	 */
+	function getDeliveryPrices($priceFormat = true, $colorFormat = true, $prefix = '', $addTax = true, $order_id = 0, $value = 0)
+	{
+		$registry = MyParcel::$registry;
+		/** @var \Cart\Currency $currency * */
+		$currency = $registry->get('currency');
+		$config = $registry->get('config');
+		$checkoutSettings = $config->get('module_myparcelnl_fields_checkout');
+
+		/**
+		 * overwrite (!) $value with total order value if $order_id is set
+		 */
+		if ($order_id > 0) {
+			$loader = $registry->get('load');
+			$loader->model('checkout/order');
+			$order = $registry->get('model_checkout_order')->getOrder($order_id);
+			$value = $order['total'];

```

Comment:
Kan het voorkomen dat de $order_id een waarde groter dan 0 heeft maar dat er geen $value wordt meegegeven in de parameter van deze functie? Want ook als de waarde van $value wordt meegegeven, als er een $order_id is wordt deze altijd overschreven. Is dat de bedoeling?

Reply:
Alleen als de $order_id niet 0 is (oftewel, als het een bestaande order is) wordt de $value overschreven met de waarde uit de order. Als er geen order meegegeven wordt, wordt gewerkt met de meegegeven waarde. Niet heel netjes, maar dit was al zo, en heb ik niet aangepast vanwege dat het maar 0,5 punt is.

Reply:
Begrijpelijk, dan zou ik het zo laten.